File: src/main/java/org/foo/dto/ClientVendorDto.java
package org.foo.dto;

import org.foo.enums.ClientVendorType;
import org.foo.models.Address;

public record ClientVendorDto(Long id,
                              String clientVendorName,
                              String phone,
                              String website,
                              ClientVendorType clientVendorType,
                              String addressLine1,
                              String addressLine2,
                              String city,
                              String state,
                              String country,
                              String zipCode,
                              CompanyDto companyDto
                              ) {
}

File: lab1-frontend/src/app/common/client-vendor-dto.ts
import {ClientVendorType} from "../enums/client-vendor-type";
import {AddressDto} from "./address-dto";
import {CompanyDto} from "./company";


export class ClientVendorDto {
  constructor(
    public id: number,
    public clientVendorName: string,
    public phone: string,
    public website: string,
    public clientVendorType: ClientVendorType,
    public addressLine1: string,
    public addressLine2: string,
    public city: string,
    public state: string,
    public country: string,
    public zipCode: string,
    public companyDto: CompanyDto
  ) {}
}
